react/components/Menu.js

```diff
@@ -0,0 +1,105 @@
+import React, { Fragment } from 'react'
+import classNames from 'classnames'
+import { path } from 'ramda'
+import PropTypes from 'prop-types'
+import { Container } from 'vtex.store-components'
+import { useRuntime } from 'vtex.render-runtime'
+
+import CategoryItem from './CategoryItem'
+import AdditionalItem from './AdditionalItem'
+import { itemPropType } from '../propTypes'
+
+import categoryMenu from '../categoryMenu.css'
+
+import categoryMenuPosition, {
+  getMenuPositionValues,
+} from '../utils/categoryMenuPosition'
+
+const Menu = ({
+  categories = [],
+  departments,
+  departmentsTitle,
+  showAllDepartments,
+  subcategoryLevels,
+  menuPosition,
+  additionalItems,
+}) => {
+  const departamentPath = path(['route', 'params', 'department'], useRuntime())
+  const fullPath = path(['route', 'path'], useRuntime())
+
+  const currentSlug = departamentPath || fullPath
+
+  const desktopClasses = classNames(
+    `${categoryMenu.container} w-100 bg-base dn flex-m`,
+    {
+      'justify-start mw9':
+        menuPosition === categoryMenuPosition.DISPLAY_LEFT.value,
+      'justify-end mw9':
+        menuPosition === categoryMenuPosition.DISPLAY_RIGHT.value,
+      'justify-center':
+        menuPosition === categoryMenuPosition.DISPLAY_CENTER.value,
+    }
+  )
+
+  return (
+    <Container className="justify-center flex">
+      <nav className={desktopClasses}>
+        <ul className="pa0 list ma0 flex flex-wrap flex-row t-action overflow-hidden h3">
+          {showAllDepartments && (
+            <CategoryItem
+              menuPosition={menuPosition}
+              subcategoryLevels={subcategoryLevels}
+              category={{
+                children: categories,
+                name: departmentsTitle,
+              }}
+            />
+          )}
+          {departments.map(category => (
+            <Fragment key={category.id}>
```

Comment:
Could remove this element and use `key` in `CategoryItem`. As `key` is a special prop, `key` won't be passed down to `CategoryItem`.